azure/services/agentpools/spec_test.go

```diff
@@ -179,15 +183,16 @@ var (
 			EnableUltraSSD:      to.BoolPtr(true),
 			MaxCount:            to.Int32Ptr(5), // updates if changed
 			MaxPods:             to.Int32Ptr(10),
-			MinCount:            to.Int32Ptr(3),                                               // updates if changed
+			MinCount:            to.Int32Ptr(1),                                               // updates if changed
```

Comment:
Each of these fixtures should be changing one field from the "default" `fakeAgentPoolSpecWithAutoscaling` defined above, but some copy-paste resulted in multiple fields being changed. The changes around here to `MinCount` and `Tags` ensure only one field is being changed at a time.

This is necessary because the new test was passing even without the fix since `MinCount` and `Tags` were triggering a diff unexpectedly that `NodeTaints` should have, but didn't.